aemedge/scripts/profile.js

```diff
@@ -0,0 +1,84 @@
+import { createOptimizedPicture, decorateIcons, toClassName } from './aem.js';
+import {
+  div, h2, p, a, span,
+} from './dom-builder.js';
+import ffetch from './ffetch.js';
+
+const breakpoints = [
+  { width: '480' },
+];
+
+function extractAuthorDescription(entry) {
+  let { description } = entry;
+  if (!description) {
+    if (entry.title) {
+      description = entry.title;
+      if (entry.author && entry.title.startsWith(`${entry.author}, `)) {
+        description = entry.title.substring(entry.author.length + 2);
+      }
+    } else {
+      description = '';
+    }
+  }
+  return description;
+}
+
+function asEntry(authorName) {
+  return { author: authorName, path: `/author/${toClassName(authorName).replace('-', '')}` };
+}
+
+function completeEntry(entry) {
+  const result = (!entry) ? { path: '/author/name', author: 'Name' } : entry;
+  result.description = extractAuthorDescription(result);
+  return result;
+}
+
+async function getAuthorEntries(keys) {
+  const entryFilter = ((entry) => (keys.includes(entry.path) || keys.includes(entry.author)));
+  const unsortedEntries = await ffetch(`${window.hlx.codeBasePath}/authors-index.json`).filter(entryFilter).limit(keys.length).all();
+  const sortedEntries = [];
+  if (unsortedEntries) {
+    keys.forEach((key) => {
+      sortedEntries.push(completeEntry(unsortedEntries.find(
+        (entry) => (key === entry.path) || key.includes(entry.author),
+      )));
+    });
+  }
+  return sortedEntries;
+}
+
+function renderProfile(entry, asAvatar = false) {
+  if (!entry) return null;
+  const authorImage = entry.image
+    ? createOptimizedPicture(entry.image, entry.author, false, breakpoints) : null;
+  const authorProfile = asAvatar
+    ? div(
+      div({ class: 'avatar' }, authorImage ? div(authorImage) : div()),
+      div(
+        span({ class: 'author' }, a({ href: entry.path }, span(`${entry.author}`))),
+        span({ class: 'info' }, entry.description),
+      ),
+    )
+    : div(
+      div({ class: 'avatar' }, authorImage ? div(authorImage) : div()),
+      div(
+        { class: 'details' },
+        h2(entry.author),
+        p(entry.description),
+        p(
+          { class: 'link' },
+          a({ href: entry.path, 'aria-label': 'Read more' }, 'See more by this author'),
+          span({ class: 'icon icon-link-arrow' }),
+        ),
+      ),
+    );
+  decorateIcons(authorProfile);
+  return authorProfile;
+}
```

Comment:
Should this be better an `authorCard` libs as well? Then we have a clear seperation between rendering and data loading (which can then be stay in the blocks itself and this file would not be needed)

Reply:
@mhaack : I propose to do the refactoring separately (e.g. in upcoming GH issue 292). My fear is this PR is growing and growing, and I would like to keep it manageable.
@saurabh-khare had also some refactoring recommendations (this is why profile.js exists in the first place).
Let's get together after March 22 to align where to best put what functionality